lib/kill.js

```diff
@@ -48,10 +49,25 @@ export const normalizeForceKillAfterDelay = forceKillAfterDelay => {
 	return forceKillAfterDelay;
 };
 
+export const waitForSuccessfulExit = async exitPromise => {
+	const [exitCode, signal] = await exitPromise;
+
+	if (!isProcessErrorExit(exitCode, signal) && isFailedExit(exitCode, signal)) {
+		const message = getErrorPrefix({signal, exitCode});
+		throw setDiscardedError(new Error(message));
```

Comment:
When `firstProcess.pipe(secondProcess)` is used and one of the process fails, we terminate the other. The second process's error message indicates it failed due to the first one. 

The second process will then use the error message being set here, to show why the first process failed.

Otherwise (when `.pipe()` is not involved), that error message is unnecessary and is discarded since `makeError()` builds it later on.